airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/DefaultJdbcDatabase.java

```diff
@@ -79,6 +81,12 @@ public DatabaseMetaData getMetaData() throws SQLException {
       final DatabaseMetaData metaData = connection.getMetaData();
       return metaData;
     } catch (final SQLException e) {
+      if (e instanceof SQLTransientException) {
+        final String message = e.getMessage();
+        if (message.contains("request timed out")) {
+          throw new ConfigErrorException("Connection timed out. Unable to contact server.");
```

Comment:
Because it's a transient error, I wonder if adding something like "Please try again later" is appropriate,
and also maybe making use of the original `e.getMessage()` can add valuable information to the user.

Reply:
> Because it's a transient error, I wonder if adding something like "Please try again later" is appropriate, and also maybe making use of the original `e.getMessage()` can add valuable information to the user.

@rodireich the connection timeout can be not only because there is some problem with the existing server, but also because if the user pushed the wrong host / port on UI which will also lead to a timeout, so just adding "Please try again later" will be not entirely correct. Might be better to use - "Please check your server settings or try again later". What do you think about it?
Replacing the original _e.getMessage()_ with a more user-friendly one is is a task of this issue. Please see this comment https://github.com/airbytehq/alpha-beta-issues/issues/331#issuecomment-1317496296

Reply:
I see. thank you for explaining!